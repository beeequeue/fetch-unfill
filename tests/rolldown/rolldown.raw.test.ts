import path from "node:path"

import { rolldown, type RolldownOptions } from "rolldown"
import { aliasPlugin } from "rolldown/experimental"
import { describe, expect, it } from "vitest"

import { createTester } from "../utils.js"

const test = createTester("rollup", async (name: string, useAlias: boolean = false) => {
  const options = {
    logLevel: "silent",
    resolve: {
      mainFields: ["module", "main"],
      conditionNames: ["node", "import", "default"],
    },
    plugins: [
      useAlias
        ? aliasPlugin({
            entries: [
              { find: "node-fetch", replacement: "fetch-unfill" },
              { find: "cross-fetch", replacement: "fetch-unfill" },
            ],
          })
        : null,
    ],
  } satisfies RolldownOptions

  const cjs = await rolldown({
    ...options,
    input: path.resolve(__dirname, "..", "fixtures", `${name}.cjs`),
  }).then(async (c) =>
    c.generate({
      format: "cjs",
      minify: true,
      inlineDynamicImports: true,
    }),
  )
  expect(cjs.output).toHaveLength(1)

  const mjs = await rolldown({
    ...options,
    input: path.resolve(__dirname, "..", "fixtures", `${name}.mjs`),
  }).then(async (c) =>
    c.generate({
      format: "esm",
      minify: true,
      inlineDynamicImports: true,
    }),
  )
  expect(mjs.output).toHaveLength(1)

  return {
    cjs: cjs.output[0].code,
    mjs: mjs.output[0].code,
  }
})

describe("node-fetch", () => {
  it("bundles the original package", async () => {
    await test("node-fetch", false)
  })

  it("unfills it", async () => {
    await test("node-fetch", true, true)
  })
})

describe("cross-fetch", () => {
  it("bundles the original package", async () => {
    await test("cross-fetch", false)
  })

  it("unfills it", async () => {
    await test("cross-fetch", true, true)
  })
})
