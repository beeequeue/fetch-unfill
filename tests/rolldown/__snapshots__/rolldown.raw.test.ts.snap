// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cross-fetch > bundles the original package > cjs 1`] = `"307.02KB"`;

exports[`cross-fetch > bundles the original package > mjs 1`] = `"310.93KB"`;

exports[`cross-fetch > unfills it > cjs 1`] = `"429B"`;

exports[`cross-fetch > unfills it > mjs 1`] = `"237B"`;

exports[`cross-fetch > unfills it > unfilled cjs code 1`] = `"var e=(e,t)=>()=>(t||e((t={exports:{}}).exports,t),t.exports),t=e((exports,t)=>{let n=globalThis.fetch;globalThis.fetch.polyfill=!1,n.polyfill=!1,t.exports=n,Object.assign(t.exports,{fetch:n,Request:globalThis.Request,Response:globalThis.Response,Headers:globalThis.Headers,AbortController:globalThis.AbortController})});const n=t();module.exports=()=>n(\`https://example.com\`),module.exports.example=()=>n(\`https://example.com\`);"`;

exports[`cross-fetch > unfills it > unfilled mjs code 1`] = `"const e=window.fetch;window.fetch.polyfill=!1,e.polyfill=!1;var t=e;const{Request:n,Response:r,Headers:i,AbortController:a}=window;var o=()=>t(\`https://example.com\`);const s=()=>t(\`https://example.com\`);export{o as default,s as example};"`;

exports[`node-fetch > bundles the original package > cjs 1`] = `"108.89KB"`;

exports[`node-fetch > bundles the original package > mjs 1`] = `"109.63KB"`;

exports[`node-fetch > unfills it > cjs 1`] = `"429B"`;

exports[`node-fetch > unfills it > mjs 1`] = `"237B"`;

exports[`node-fetch > unfills it > unfilled cjs code 1`] = `"var e=(e,t)=>()=>(t||e((t={exports:{}}).exports,t),t.exports),t=e((exports,t)=>{let n=globalThis.fetch;globalThis.fetch.polyfill=!1,n.polyfill=!1,t.exports=n,Object.assign(t.exports,{fetch:n,Request:globalThis.Request,Response:globalThis.Response,Headers:globalThis.Headers,AbortController:globalThis.AbortController})});const n=t();module.exports=()=>n(\`https://example.com\`),module.exports.example=()=>n(\`https://example.com\`);"`;

exports[`node-fetch > unfills it > unfilled mjs code 1`] = `"const e=window.fetch;window.fetch.polyfill=!1,e.polyfill=!1;var t=e;const{Request:n,Response:r,Headers:i,AbortController:a}=window;var o=()=>t(\`https://example.com\`);const s=()=>t(\`https://example.com\`);export{o as default,s as example};"`;
